### DISCUSSION TITLE
MODELS PATH

### DISCUSSION THREAD

**Discussion by sana3dh** [suno-ai/bark#251] [Q&A] [answered]
i have **cloned the Repo** and **installed requirements** but when i want to run a test , it **starts to download models** , 
i have a **bad internet connection** so i **downloaded all the models manually** but i don't know how to do this and **where to put them**
anyone can guide me how to do that ?
_(windows 10 / CPU mode / Manually Downloaded :text.pt,coarse.pt,fine.pt,text_2.pt,coarse_2.pt,fine_2.pt)_

**Comment by gkucsko** [comment, marked as answer]
here is the relevant code:
```
default_cache_dir = os.path.join(os.path.expanduser("~"), ".cache")
CACHE_DIR = os.path.join(os.getenv("XDG_CACHE_HOME", default_cache_dir), "suno", "bark_v0")
```
so most likely they should be in `~/.cache/suno/bark_v0/` however huggingface does some magic in the naming so not entirely sure how it's saved, we just updated to their hub sorry. If anyone has experience would be super helpful to paste a short 'manual model install' here

**Reply by zxg1024** [reply]
Changing default_cache_dir does change the location of models--suno--bark, but the locations of models--bert-base-multilingual-cased and encodec_24khz-d7cc33bc.th remain in /.cache

**Reply by zxg1024** [reply]
Mobile encodec_24khz-d7cc33bc.th:

1. Install pathlib and add it to the generation.py

`from pathlib import Path`

2. Modify here:

```
# model = EncodecModel.encodec_model_24khz()
model = EncodecModel.encodec_model_24khz(repository = Path(default_cache_dir))
```

3. Move the encodec_24khz-d7cc33bc.th file in .cache/torch/hub/checkpoints/ to the default_cache_dir

**Comment by AK391** [comment]
also check out https://huggingface.co/docs/huggingface_hub/guides/download, and you can go through the UI as well to download files individually

**Comment by wp337** [comment]
Where did you download the model from? My network is also not good, I also need to download the model

**Reply by sana3dh** [reply]
i downloaded them weeks ago so they may have changed ... :
https://dl.suno-models.io/bark/models/v0/fine.pt
https://dl.suno-models.io/bark/models/v0/fine_2.pt
https://dl.suno-models.io/bark/models/v0/text.pt
https://dl.suno-models.io/bark/models/v0/text_2.pt
https://dl.suno-models.io/bark/models/v0/coarse.pt
https://dl.suno-models.io/bark/models/v0/coarse_2.pt

**Comment by wp337** [comment]
thank you for you help bro,my english is bed,have nice day



---Original---
From: ***@***.***&gt;
Date: Wed, May 17, 2023 01:41 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [suno-ai/bark] MODELS PATH (Discussion #251)




 
i downloaded them weeks ago so they may have changed ... :
 https://dl.suno-models.io/bark/models/v0/fine.pt
 https://dl.suno-models.io/bark/models/v0/fine_2.pt
 https://dl.suno-models.io/bark/models/v0/text.pt
 https://dl.suno-models.io/bark/models/v0/text_2.pt
 https://dl.suno-models.io/bark/models/v0/coarse.pt
 https://dl.suno-models.io/bark/models/v0/coarse_2.pt

**Comment by mike2505** [comment]
Not sure where should i put downloaded models.

ls: blobs  refs  snapshots
pwd: /home/USER/.cache/huggingface/hub/models--suno--bark

**Reply by zxg1024** [reply]
Look at this line in generation.py file

`CACHE_DIR = os.path.join(os.getenv("XDG_CACHE_HOME", default_cache_dir), "suno", "bark_v0")`

could be a directory spelled out of these
It could be inside .cache/suno/bark_v0
